homeassistant/components/nanoleaf/config_flow.py

```diff
@@ -0,0 +1,226 @@
+        return await self.async_setup_finish()
+
+    async def async_setup_finish(self) -> FlowResult:
+        """Finish Nanoleaf config flow."""
+        try:
+            info = await self.hass.async_add_executor_job(
+                pynanoleaf_get_info, self.nanoleaf
+            )
+        except Unavailable:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=USER_SCHEMA,
+                errors={"base": "cannot_connect"},
+                last_step=False,
+            )
+        except InvalidToken:
+            return self.async_show_form(
+                step_id="link", errors={"base": "invalid_token"}
+            )
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unknown error connecting with Nanoleaf at %s with token %s",
+                self.nanoleaf.host,
+                self.nanoleaf.token,
+            )
+            return self.async_show_form(
+                step_id="user",
+                data_schema=USER_SCHEMA,
+                errors={"base": "unknown"},
+                last_step=False,
+            )
+        name = info["name"]
+
+        await self.async_set_unique_id(name)
+        self._abort_if_unique_id_configured({CONF_HOST: self.nanoleaf.host})
+
+        # Show successfully imported instructions and optionally update the Nanoleaf config file
+        if self.context["source"] == config_entries.SOURCE_IMPORT:
+            persistent_notification.async_create(
+                self.hass,
+                f"Nanoleaf {name} has been successfully imported. Please remove the light with host {self.nanoleaf.host} from your configuration.yaml.",
+                f"Imported Nanoleaf {name}",
+            )
+        elif self.context["source"] == config_entries.SOURCE_INTEGRATION_DISCOVERY:
+            if self.nanoleaf.host in self.discovery_conf:
+                self.discovery_conf.pop(self.nanoleaf.host)
+            if self.device_id in self.discovery_conf:
+                self.discovery_conf.pop(self.device_id)
+            _LOGGER.info(
+                "Successfully imported Nanoleaf %s from the discovery integration",
+                name,
+            )
+            if self.discovery_conf:
+                await self.hass.async_add_executor_job(
+                    save_json, self.hass.config.path(CONFIG_FILE), self.discovery_conf
+                )
+            else:
+                await self.hass.async_add_executor_job(
+                    os.remove, self.hass.config.path(CONFIG_FILE)
+                )
+                persistent_notification.async_create(
+                    self.hass,
+                    "All Nanoleaf devices from the discovery integration are imported. If you used the discovery integration only for Nanoleaf you can remove it from your configuration.yaml",
+                    f"Imported Nanoleaf {name}",
+                )
```

Comment:
```suggestion
```

We generally don't generate persistent notification on config entry import.

Reply:
I was told to use a persistent notification instead of logging. https://github.com/home-assistant/core/pull/52199#discussion_r660270925
Integrations usually log in setup_platforms. This does not work for the discovery integration users. A notification is better because we can only show this message once after all devices are imported.

Reply:
Usually we won't want a notification if action isn't required by the user.

This case looks like it could be the exception since action might be required by the user. 👍